app/services/NotificationLists.scala

```diff
@@ -41,6 +47,35 @@ object NotificationLists {
     case _ => Either.right(atom)
   }
 
+  def sendNotificationList(atom: Atom): Either[AtomAPIError, Atom] = atom.data match {
+    case AtomData.Storyquestions(data) => {
+      val result = for {
+        userQuestions <- data.editorialQuestions
+        questionSet <- userQuestions.headOption //We only use the first questionSet in the atom
+      } yield {
+        val results = for {
+          question <- questionSet.questions
+          if question.answers.nonEmpty
+        } yield {
+          val qa = QuestionAnswers.fromThrift(atom.id, question)
+          var request = new InvokeRequest()
+            .withFunctionName(Config.lambdaFunctionName)
+            .withPayload(qa.asJson.toString)
+          lambdaClient.invoke(request)
+        }
+  
+        if (results.forall(_.getStatusCode == 200))
+          Either.right(atom)
+        else
+          Either.left(LambdaExecutionError(Config.lambdaFunctionName, "Unable to trigger email notifications"))
```

Comment:
I think that it would be useful to add the error status code (and possibly the body if it's not a security issue) to the `LambdaExecutionError`. It can help with debugging.

Another curious question: Doesn't this app use logging?

Reply:
logging does seem quite limited in this app, though it must log the errors - https://github.com/guardian/atom-workshop/blob/021c33a0e87161621060860baf6290d5719bba0d/app/models/APIResponse.scala#L16